### ISSUE TITLE
Yolo Assertion: l.outputs == params.inputs' failed.

### ISSUE THREAD

**Issue by Arktius** [pjreddie/darknet#1637]
Hej,

I'm struggling with the number of filters in the .cfg file. I've seen different formulas, but non of them is working. 

This is my file:

[net]
batch=2
subdivisions=1
width=416#900#416
height=416#900#416
channels=3
momentum=0.9
decay=0.0005

learning_rate= 0.001
burn_in=800
max_batches = 10000
policy=steps
steps=4000,7500
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=21
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=99
activation=linear

[yolo]
mask = 3,4,5
#anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
anchors = 9,18,   11,31,  13,9,  15,85,  16,14,  18,29
classes=28
num=6
#jitter=.3
ignore_thresh = .55
truth_thresh = 1
#random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=99
activation=linear

[yolo]
mask = 0,1,2 #1,2,3
#anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
anchors = 9,18,   11,31,  13,9,  15,85,  16,14,  18,29
classes=28
num=6
#jitter=.3
ignore_thresh = .55
truth_thresh = 1
#random=1

I have 2 yolo-layers, num = 6 and 28 classes. filters = (5+classes)x(num/yolo-layers) = 33 x 3 = 99.

Another question:

Can I easily change the width and height? What effects does it have? Do I need to change something else? My images are way bigger than the used training set and the targets quite small additionally.

**Comment by dsgh2** [comment]
Please check your .data file whether that of class number is same to your .cfg

**Comment by Arktius** [comment]
obj28.data:

classes = 28
train = data/train.txt
valid = data/test.txt
names = data/obj.names
backup = backup/

It's also 28. There are any other possibilities to get this error?

**Comment by Arktius** [comment]
I've found the mistake. Mask 0 doesn't exist. It's 1,..., num.

**Comment by philipperemy** [comment]
Just for information, I was not updating the correct convolutional layers.

https://www.ccoderun.ca/programming/2019-08-18_Training_with_Darknet/

I have to read more carefully.
